components/gitpod-db/src/typeorm/installation-admin-db-impl.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { inject, injectable, } from 'inversify';
+import { InstallationAdmin } from '@gitpod/gitpod-protocol';
+import { v4 as uuidv4 } from 'uuid';
+import { Repository } from 'typeorm';
+import { TypeORM } from './typeorm';
+import { InstallationAdminDB } from '../installation-admin-db';
+import { DBInstallationAdmin } from './entity/db-installation-admin';
+
+@injectable()
+export class TypeORMInstallationAdminImpl implements InstallationAdminDB {
+    @inject(TypeORM) typeORM: TypeORM;
+
+    protected async getEntityManager() {
+        return (await this.typeORM.getConnection()).manager;
+    }
+
+    protected async createDefaultRecord(): Promise<InstallationAdmin> {
+        const record: InstallationAdmin = {
+            id: uuidv4(),
```

Comment:
Can/should we use this ID as installation ID and replace the domain hash that we currently use as installation ID? Once generated and saved to the DB, this ID shouldn't change anymore, right? What do you think?

Reply:
It's not impossible for the installation ID to change - if someone changes their domain and/or changes their pepper (see #7538), this will change the installation ID. Personally, I feel it's better to decouple the installation ID and the DB record ID